src/main/resources/std/string.lua

```diff
@@ -1,114 +1,138 @@
--- Copyright (c) 2017. tangzx(love.tangzx@qq.com)
+-- Copyright (c) 2018. tangzx(love.tangzx@qq.com)
 --
--- Licensed under the Apache License, Version 2.0 (the "License");
--- you may not use this file except in compliance with the License.
--- You may obtain a copy of the License at
+-- Licensed under the Apache License, Version 2.0 (the "License"); you may not 
+-- use this file except in compliance with the License. You may obtain a copy of
+-- the License at
 --
 -- http://www.apache.org/licenses/LICENSE-2.0
 --
--- Unless required by applicable law or agreed to in writing, software
--- distributed under the License is distributed on an "AS IS" BASIS,
--- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
--- See the License for the specific language governing permissions and
--- limitations under the License.
+-- Unless required by applicable law or agreed to in writing, software 
+-- distributed under the License is distributed on an "AS IS" BASIS, WITHOUT 
+-- WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the 
+-- License for the specific language governing permissions and limitations under
+-- the License.
 
 ---@class string
 string = {}
 
 ---
 --- Returns the internal numerical codes of the characters `s[i]`, `s[i+1]`,
 --- ..., `s[j]`. The default value for `i` is 1; the default value for `j`
---- is `i`.
+--- is `i`. These indices are corrected following the same rules of function
+--- `string.sub`.
+---
 --- Note that numerical codes are not necessarily portable across platforms.
 ---@param s string
 ---@param optional i number
 ---@param optional j number
+---@return number
 function string.byte(s, i, j) end
 
 ---
 --- Receives zero or more integers. Returns a string with length equal to
 --- the number of arguments, in which each character has the internal numerical
 --- code equal to its corresponding argument.
+---
 --- Note that numerical codes are not necessarily portable across platforms.
 ---@return string
 function string.char(...) end
 
 ---
---- Returns a string containing a binary representation of the given
---- function, so that a later `loadstring` on this string returns a copy of
---- the function. `function` must be a Lua function without upvalues.
+--- Returns a string containing a binary representation (*a binary chunk*) of
+--- the given function, so that a later `load` on this string returns a
+--- copy of the function (but with new upvalues). If strip is a true value, the 
+--- binary representation may not include all debug information about the 
+--- function, to save space.
+---
+--- Functions with upvalues have only their number of upvalues saved. When (re)
+--- loaded, those upvalues receive fresh instances containing **nil**. (You can 
+--- use the debug library to serialize and reload the upvalues of a function in 
+--- a way adequate to your needs.)
 ---@param func fun()
-function string.dump(func) end
+---@param optional strip boolean
```

Comment:
`optional` is deprecated. use `@overload` instead of.

```
---@overload fun(func:fun()):string
---@param func fun()
---@param strip boolean
---@return string
function string.dump(func, strip) end
```

Reply:
Ow, okay! I'll change the appearance of optional in the code. What about the types. Do they need to stay in small caps?

Reply:
Yes,  the built-in (primitive) types written in lower case, just like c#/as3/...

Reply:
Okay, C# also supports OOP types - in CamelCase, but I'll leave it all in lower.